### PR TITLE
BLO-27 Query filtering upgraded

### PR DESCRIPTION
**Pull request by Svenmarim**

### CODE REVIEW

**Review by ronaldvandenbroek — APPROVED** [review]

File: src/repositories/concrete/mongoTransactionRepository.ts [hunk 1]
```diff
@@ -1,32 +1,32 @@
-import { Transaction } from "@blockr/blockr-models";
+import { Transaction, TransactionType } from "@blockr/blockr-models";
```
Comment (Juuldamen) [inline src/repositories/concrete/mongoTransactionRepository.ts:1]: Put these imports in alphabetical order

**Review by Juuldamen — APPROVED** [review]

File: src/exceptions/fieldDoesNotExist.exeception.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export class FieldDoesNotExist extends Error {
```
Comment (timofudala) [inline src/exceptions/fieldDoesNotExist.exeception.ts]: The class name should actually be `FieldDoesNotExistException`.
Please also mind the fact that there is a typo in the file name. It should be `fieldDoesNotExist.exception.ts`, mind the `.exception` part.

File: src/repositories/concrete/mongoDbQueryBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                        finalQuery.push(filter);
+                        break;
+                    case "String":
+                        filter[queryKey] = urlParams[queryKey];
+                        finalQuery.push(filter);
+                        break;
+                    default:
+                        throw new TypeNotImplementedException(`Type: ${type} has not been implemented on: ${queryKey}`);
+                }
+            } catch (err) {
```
Comment (timofudala) [inline src/repositories/concrete/mongoDbQueryBuilder.ts]: please change `err` to `error`.

File: src/repositories/concrete/mongoDbQueryBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                throw new FieldDoesNotExist(`Field: ${queryKey} does not exist on: ${exampleModel.constructor.name}`);
+            }
+        }
+
+        return finalQuery.length < 1 ? {} : { $and: finalQuery };
```
Comment (timofudala) [inline src/repositories/concrete/mongoDbQueryBuilder.ts]: I'm nitpicking, but please change `{}` to `{ }`.

Please also consider changing `finalQuery.length < 1` to `finalQuery.length == 0`.

**Review by timofudala — CHANGES_REQUESTED** [review]

**Review by timofudala — APPROVED** [review]

**Pull request merged**